pinot-tools/src/main/resources/examples/batch/airlineStats/airlineStats_offline_table_config.json

```diff
@@ -23,6 +23,33 @@
           "MONTH"
         ]
       }
+    },
+    {
+      "name": "ArrTimeBlk",
+      "encodingType": "DICTIONARY",
+      "indexes": {
+        "inverted": {
+          "enabled": "true"
+        }
+      },
+      "tierOverwrites": {
+        "hotTier": {
+          "encodingType": "DICTIONARY",
+          "indexes": {
+            "bloom": {
+              "enabled": "true"
+            }
+          }
+        },
+        "coldTier": {
+          "encodingType": "RAW",
```

Comment:
Reading the code I thought we weren't going to support overriding encoding type. Which part of the code actually overrides the effective encodingType? For example ForwardIndexType#137 is still reading the default encoding type, ignoring the override. Same with Dictionary.

Reply:
It's allowed to overwrite encoding type, as done by `FieldConfig.withTierOverwrites(tier)` method, where the tier specific FieldConfig object is constructed and may use tier specific encoding type, indexes or properties.

DictionaryIndexType has this`fieldConfig = fieldConfig.withTierOverwrites(tier);`, but I missed it for ForwardIndexType.